activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -366,10 +366,15 @@ def type_cast_calculated_value(value, type, operation = nil)
       end
     end
 
-    # TODO: refactor to allow non-string `select_values` (eg. Arel nodes).
     def select_for_count
       if select_values.present?
-        select_values.join(", ")
+        select_values.map do |field|
```

Comment:
@rafaelfranca That's exactly the path I'm following now, but it'll take me some days to have something to push.

Let's see what @sgrif thinks.